### PR TITLE
Merge client.{kv,txn} into client.{DB,Tx}.

### PR DESCRIPTION
**Pull request by petermattis**
Still todo: rename client.Tx to client.Txn.

See #997.

### CODE REVIEW

File: client/txn.go [hunk 1]
```diff
@@ -77,30 +85,192 @@ type txn struct {
...
+// function returns without error.
+func (tx *Tx) Commit(b *Batch) error {
+	args := &proto.EndTransactionRequest{Commit: true}
+	reply := &proto.EndTransactionResponse{}
+	b.calls = append(b.calls, Call{Args: args, Reply: reply})
+	b.initResult(1, 0, nil)
+	return tx.Run(b)
+}
+
+func (tx *Tx) init(db *DB) {
```
Comment (tamird) [inline client/txn.go]: this function :crying_cat_face:

Comment (petermattis) [inline client/txn.go]: Yeah, I did this originally when I didn't want `txn` to be allocated separately from `Tx`, but now that they are merged adding a `newTx` function seems better.

Comment (petermattis) [inline client/txn.go]: Ok, done. Definitely better with a `newTx` function.

File: client/txn.go [hunk 2]
```diff
@@ -77,30 +85,200 @@ type txn struct {
...
-	t.wrapped = kv.Sender
-	t.txnSender.txn = t
-	t.kv.Sender = &t.txnSender
+func newTx(db *DB, depth int) *Tx {
+	tx := &Tx{
+		db:      *db,
+		wrapped: db.Sender,
+	}
+	tx.txnSender.Tx = tx
+	tx.db.Sender = &tx.txnSender
```
Comment (tamird) [inline client/txn.go]: if the db's Sender is equal to tx's txnsender, why have both? Tx could have a getter that fetches from db, no?

Comment (petermattis) [inline client/txn.go]: `Tx.txnSender` is the transaction specific sender. A `Tx` internally uses a `DB` in order to send calls using `DB.send`, but it wants that `send` to use the transaction sender. See `DB.send`.

**Comment by petermattis** [comment]
Looking for that LGTM. I can walk you through the code structure if you'd like, but note that this change is essentially only code movement.

File: client/txn.go (hunk 2 above)
Comment (tschottdorf) [inline client/txn.go]: Would love to see this simplified in a subsequent change. TODO maybe?

**Comment by tschottdorf** [comment]
LGTM

**Pull request merged**